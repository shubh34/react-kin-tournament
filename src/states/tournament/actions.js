import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';
import { cacheApiResponse } from '../../utils/cacheApiUtils';

export const GET_TOURNAMENT_LEADERS_REQUEST = 'GET_TOURNAMENT_LEADERS_REQUEST';
export const GET_TOURNAMENT_LEADERS_SUCCESS = 'GET_TOURNAMENT_LEADERS_SUCCESS';
export const GET_TOURNAMENT_LEADERS_FAILURE = 'GET_TOURNAMENT_LEADERS_FAILURE';

export const GET_TOURNAMENT_PRIZES_REQUEST = 'GET_TOURNAMENT_PRIZES_REQUEST';
export const GET_TOURNAMENT_PRIZES_SUCCESS = 'GET_TOURNAMENT_PRIZES_SUCCESS';
export const GET_TOURNAMENT_PRIZES_FAILURE = 'GET_TOURNAMENT_PRIZES_FAILURE';

export const createFetchLeaderRSSAAction = () => {
	const endPoint = endPoints.getLeaderBoard;
	return {
		[RSAA]: {
			endpoint: endPoint.url,
			method: endPoint.method,
			types: [
				GET_TOURNAMENT_LEADERS_REQUEST,
				{
					type: GET_TOURNAMENT_LEADERS_SUCCESS,
					meta: {
						cacheKey: GET_TOURNAMENT_LEADERS_REQUEST,
					},
				},
				GET_TOURNAMENT_LEADERS_FAILURE,
			],
		},
	};
};

export const createFetchPrizesRSSAAction = () => {
	const endPoint = endPoints.fetchTournamentPrizes;
	return {
		[RSAA]: {
			endpoint: endPoint.url,
			method: endPoint.method,
			types: [
				GET_TOURNAMENT_PRIZES_REQUEST,
				{
					type: GET_TOURNAMENT_PRIZES_SUCCESS,
					meta: {
						cacheKey: GET_TOURNAMENT_PRIZES_REQUEST,
					},
				},
				GET_TOURNAMENT_PRIZES_FAILURE,
			],
		},
	};
}
export const fetchTournamentLeaders = () => (dispatch) => {
	const fetchLeaderRSSAAction = createFetchLeaderRSSAAction();
	return dispatch(fetchLeaderRSSAAction).then((response) => {
		cacheApiResponse(response);
	});
};

export const fetchTournamentPrizes = () => (dispatch) => {
	const fetchPrizesRSSAAction = createFetchPrizesRSSAAction();
	return dispatch(fetchPrizesRSSAAction).then((response) => {
		cacheApiResponse(response);
	});
};
