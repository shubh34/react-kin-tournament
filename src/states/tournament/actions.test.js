import { fetchTournamentLeaders, fetchTournamentPrizes } from './actions';

describe('Tournament actions', () => {
	it('should create action to fetch leader', () => {
		expect(fetchTournamentLeaders()).toMatchSnapshot();
	});
	it('should create action to fetch prizes', () => {
		expect(fetchTournamentPrizes()).toMatchSnapshot();
	});
})
;