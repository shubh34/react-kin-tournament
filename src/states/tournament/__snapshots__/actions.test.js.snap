// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tournament actions should create action to fetch leader 1`] = `
Object {
  "@@redux-api-middleware/RSAA": Object {
    "endpoint": "https://tournament-api-87.herokuapp.com/leaderboard",
    "method": "GET",
    "types": Array [
      "GET_TOURNAMENT_LEADERS_REQUEST",
      "GET_TOURNAMENT_LEADERS_SUCCESS",
      "GET_TOURNAMENT_LEADERS_FAILURE",
    ],
  },
}
`;

exports[`Tournament actions should create action to fetch prizes 1`] = `
Object {
  "@@redux-api-middleware/RSAA": Object {
    "endpoint": "https://tournament-api-87.herokuapp.com/prizes",
    "method": "GET",
    "types": Array [
      "GET_TOURNAMENT_PRIZES_REQUEST",
      "GET_TOURNAMENT_PRIZES_SUCCESS",
      "GET_TOURNAMENT_PRIZES_FAILURE",
    ],
  },
}
`;
